### PR TITLE
114 circuit representation of operation history in cli

### LINKED ISSUE #114: Circuit representation of operation history in CLI

### PR DESCRIPTION
**Pull request by BAMF0**
This will add the ability to show the application history as a circuit.
Furthermore, this will also refactor the CLI module to be more lean.

Most functions and types have been moved out to the `cli_utils` module instead.

### CODE REVIEW

**Review by edvinsternvik — APPROVED** [review]
Overall it look great, it is a large change and would probably have been easier to look through if it was split into smaller changes.

The code looks great and the circuit visualization looks nice and is definitely a positive change for the project.

I noticed that it doesn't work for OpenQASM programs which makes sense since `run_openqasm` only returns the registers - and not the operations - which makes it impossible to implement the history. So implementing this functionality is probably out of scope for this PR.

That was the only issue I found, which I will create a separate issue for.

TLDR: Look great :)

**Comment by simonengstrm** [comment]
@edvinsternvik Updated the branch, could you approve the changes again?
